sys/clang-format.py

```diff
@@ -34,27 +36,50 @@
     "librz/util/bdiff.c",
 ]
 
-pattern = ["*.c", "*.cpp", "*.h", "*.hpp", "*.inc"]
+patterns = ["*.c", "*.cpp", "*.h", "*.hpp", "*.inc"]
 
 
 def skip(filename):
-    for s in skiplist:
-        if s in filename:
-            return True
-    return False
-
-
-try:
-    for d in dirlist:
-        print("Processing directory: {0}".format(d))
-        for pat in pattern:
-            print("Processing pattern: {0}".format(pat))
-            for filename in glob.iglob(d + "/**/" + pat, recursive=True):
-                if not skip(filename):
-                    CMD = "clang-format -style=file -i {0}".format(filename)
-                    print(CMD)
-                    os.system(CMD)
-
-except KeyboardInterrupt:
-    print("clang-format.py interrupted by the user.")
-    sys.exit(1)
+    return any(skipfile in filename for skipfile in skiplist)
+
+
+def build_command(ci, filename):
+    if ci:
+        return "clang-format -style=file --Werror --dry-run {0}".format(filename)
+
+    return "clang-format -style=file -i {0}".format(filename)
+
+
+def get_matching_files():
+    for directory, pattern in itertools.product(dirlist, patterns):
+        for filename in glob.iglob(directory + "/**/" + pattern, recursive=True):
+            if not skip(filename):
+                yield filename
+
+
+def format_rizin(args):
+    ret = 0
+
+    for filename in get_matching_files():
+        cmd = build_command(args.ci, filename)
+        if args.verbose:
+            print(cmd)
+        if os.system(cmd):
+            ret = 1
+
+    sys.exit(ret)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Clang format the rizin project")
+    parser.add_argument(
+        "--ci", action="store_true", help="Flag that enable the CI mode"
```

Comment:
Better call it something like `--check` instead of `--ci` to indicate what it actually does. In addition, I think this can also be useful outside of the CI.